cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala

```diff
@@ -69,15 +72,32 @@ class UsingTest extends DocumentingTest {
       }
     }
     section("Index hints", "query-using-index-hint") {
-      p("""Index hints are used to specify which index, the planner should use as a starting point.
+      p("""Index hints are used to specify which index the planner should use as a starting point.
           |This can be beneficial in cases where the index statistics are not accurate for the specific values that
           |the query at hand is known to use, which would result in the planner picking a non-optimal index.
-          |To supply an index hint, use `USING INDEX variable:Label(property)` or `USING INDEX SEEK variable:Label(property)` after the applicable `MATCH` clause for node indexes,
-          |and `USING INDEX variable:RELATIONSHIP_TYPE(property)` or `USING INDEX SEEK variable:RELATIONSHIP_TYPE(property)` for relationship indexes.""")
+          |An index hint is supplied after an applicable `MATCH` clause. Available index hints are:""")
       p(
-        """`USING INDEX` can be fulfilled by any of the following plans:
-          |`NodeIndexScan`, `DirectedRelationshipIndexScan`, `UndirectedRelationshipIndexScan`, `NodeIndexSeek`, `DirectedRelationshipIndexSeek`, `UndirectedRelationshipIndexSeek`.
-          |`USING INDEX SEEK` can only be fulfilled by `NodeIndexSeek`, `DirectedRelationshipIndexSeek` or `UndirectedRelationshipIndexSeek`.""")
+        """
+          |[options="header"]
+          ||===
+          || Hint                                                                     | Fulfilled by plans
+          || `USING [BTREE \| TEXT] INDEX variable:Label(property)`                   | `NodeIndexScan`, `NodeIndexSeek`
+          || `USING [BTREE \| TEXT] INDEX SEEK variable:Label(property)`              | `NodeIndexSeek`
+          || `USING [BTREE \| TEXT] INDEX  variable:RELATIONSHIP_TYPE(property)`      | `DirectedRelationshipIndexScan`, `UndirectedRelationshipIndexScan`, `DirectedRelationshipIndexSeek`, `UndirectedRelationshipIndexSeek`
+          || `USING [BTREE \| TEXT] INDEX SEEK variable:RELATIONSHIP_TYPE(property)`  | `DirectedRelationshipIndexSeek`, `UndirectedRelationshipIndexSeek`
+          ||===
+          """
+      )
+      p(
+        """When specifying an index type for a hint, e.g. `BTREE` or `TEXT`, the hint can only be fulfilled when an index of the specified type is available.
+          |When no index type is specified, the hint can be fulfilled by any index types.
+          |""".stripMargin)
+      caution {
+        p(
+          """Using an index hint will never change the result of a query.
```

Comment:
I think this can be generalized:
```suggestion
          """Using a hint must never change the result of a query.
```